etc/system-contracts/contracts/DefaultAccountNoSecurity.sol

```diff
@@ -0,0 +1,210 @@
+        if (codeAddress != address(this)) {
+            // If the function was delegate called, behave like an EOA.
+            assembly {
+                return(0, 0)
+            }
+        }
+
+        // Continue execution if not delegate called.
+        _;
+    }
+
+    /// @notice Validates the transaction & increments nonce.
+    /// @dev The transaction is considered accepted by the account if
+    /// the call to this function by the bootloader does not revert
+    /// and the nonce has been set as used.
+    /// @param _suggestedSignedHash The suggested hash of the transaction to be signed by the user.
+    /// This is the hash that is signed by the EOA by default.
+    /// @param _transaction The transaction structure itself.
+    /// @dev Besides the params above, it also accepts unused first paramter "_txHash", which
+    /// is the unique (canonical) hash of the transaction.
+    function validateTransaction(
+        bytes32, // _txHash
+        bytes32 _suggestedSignedHash,
+        Transaction calldata _transaction
+    ) external payable override ignoreNonBootloader ignoreInDelegateCall returns (bytes4 magic) {
+        magic = _validateTransaction(_suggestedSignedHash, _transaction);
+    }
+
+    /// @notice Inner method for validating transaction and increasing the nonce
+    /// @param _suggestedSignedHash The hash of the transaction signed by the EOA
+    /// @param _transaction The transaction.
+    function _validateTransaction(
+        bytes32 _suggestedSignedHash,
+        Transaction calldata _transaction
+    ) internal returns (bytes4 magic) {
+        // Note, that nonce holder can only be called with "isSystem" flag.
+        SystemContractsCaller.systemCallWithPropagatedRevert(
+            uint32(gasleft()),
+            address(NONCE_HOLDER_SYSTEM_CONTRACT),
+            0,
+            abi.encodeCall(INonceHolder.incrementMinNonceIfEquals, (_transaction.nonce))
+        );
+
+        // Even though for the transaction types present in the system right now,
+        // we always provide the suggested signed hash, this should not be
+        // always expected. In case the bootloader has no clue what the default hash
+        // is, the bytes32(0) will be supplied.
+        bytes32 txHash = _suggestedSignedHash != bytes32(0) ? _suggestedSignedHash : _transaction.encodeHash();
+
+        // The fact there is are enough balance for the account
+        // should be checked explicitly to prevent user paying for fee for a
+        // transaction that wouldn't be included on Ethereum.
+        uint256 totalRequiredBalance = _transaction.totalRequiredBalance();
+        require(totalRequiredBalance <= address(this).balance, "Not enough balance for fee + value");
+
+        if (_isValidSignature(txHash, _transaction.signature)) {
+            magic = ACCOUNT_VALIDATION_SUCCESS_MAGIC;
+        } else {
+            magic = bytes4(0);
+        }
+    }
```

Comment:
Since we are returning `true` for `_isValidSignature` can't we minimize this to the below snippet? Or do we want to maintain consistency?

```solidity
function _validateTransaction(
        bytes32 _suggestedSignedHash,
        Transaction calldata _transaction
    ) internal returns (bytes4 magic) {
        // Note, that nonce holder can only be called with "isSystem" flag.
        SystemContractsCaller.systemCallWithPropagatedRevert(
            uint32(gasleft()),
            address(NONCE_HOLDER_SYSTEM_CONTRACT),
            0,
            abi.encodeCall(INonceHolder.incrementMinNonceIfEquals, (_transaction.nonce))
        );

        // The fact there is are enough balance for the account
        // should be checked explicitly to prevent user paying for fee for a
        // transaction that wouldn't be included on Ethereum.
        uint256 totalRequiredBalance = _transaction.totalRequiredBalance();
        require(totalRequiredBalance <= address(this).balance, "Not enough balance for fee + value");

       magic = ACCOUNT_VALIDATION_SUCCESS_MAGIC;
    }
```